include/oneapi/dpl/internal/distributed_ranges_impl/concepts.hpp

```diff
@@ -0,0 +1,84 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_DR_CONCEPTS_CONCEPTS_HPP
+#define _ONEDPL_DR_CONCEPTS_CONCEPTS_HPP
+
+#include "detail/ranges.hpp"
+
+namespace oneapi::dpl::experimental::dr
+{
+
+template <typename I>
+concept remote_iterator = std::forward_iterator<I> && requires(I& iter)
+{
+    ranges::rank(iter);
```

Comment:
can we potentially end up with the name conflict for `ranges` namespace name? My guess is we should not based on my knowledge of lookup rules but do we want to change this line and the lines below to `dr::ranges` just to be safe?

To be honest I don't have a strong preference